jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/HttpChannelOverHTTP2.java

```diff
@@ -285,6 +285,23 @@ public void recycle()
         getHttpTransport().recycle();
     }
 
+    public void onAsyncWaitForContent()
+    {
+        // Check if demand was fulfilled while we were transitioning to the REGISTERED state,
+        // push the content to the HttpInput if that was the case.
+        boolean[] endStreamResult = new boolean[1];
+        HttpInput.Content content = _requestContent.takeContent(endStreamResult);
```

Comment:
Considering that HttpInput.Content is our interface, why don't we just put an `isLast()` method on it?

Reply:
This implementation conflates too many things and needs to be simplified. See my latest version.